aepsych/server/server.py

```diff
@@ -334,13 +334,27 @@ def handle_setup_v01(self, request):
             experiment_id = None
             if self._db_master_record is not None:
                 experiment_id = self._db_master_record.experiment_id
-
-            self._db_master_record = self.db.record_setup(
-                description="default description",
-                name="default name",
-                request=request,
-                id=experiment_id,
-            )
+            ### make a temporary config object to derive parameters because server handles config after table
+            tempconfig = Config(**request["message"])
```

Comment:
Why not modify `self.configure` to take in a `Config` object instead of the raw message? Then you don't need to create it twice (the first thing `self.configure` does is create a `Config` anyway).

Reply:
Hey @mshvartsman ! I absolutely agree in that it would be a great idea to just pass the config object. One thing I am concerned about is that self.configure is also called by the legacy `self.configure` function with a string. I would have to write another version of this function to adjust to the new `handle_setup_v01` but again it is important to think about whether you'd like to see another secondary version of an existing function. Please let me know if you'd like me to proceed. (TLDR: I don't want to modify the original `handle_setup` in any way unless absolutely necessary).

Reply:
I think modifying the original `handle_setup` is maybe okay here. To the extent that there's a contract here it's that legacy `handle_setup`'s behavior doesn't change, which will still be true if you change where the string-to-config conversion happens. I don't think anything else is calling `self.configure`. @crasanders what do you think?

Reply:
Yeah, I think it's fine to change handle_setup to pass a `Config` into `self.configure`.

Reply:
Okie dokie, will do!